### ISSUE TITLE
Potential incorrect bucketing of API rate limit

### ISSUE THREAD

**Issue by ymylei** [okta/terraform-provider-okta#913]
https://github.com/okta/terraform-provider-okta/blob/1897d8d932e6af2b6a394c764f725670934227d1/okta/internal/apimutex/apimutex_test.go#L123

This indicates that the users/groups calls to specific apps is rate limited at the `api/v1/apps` bucket instead of the `api/v1/apps/${id}` bucket.

Based on my reading of the RL docs (https://developer.okta.com/docs/reference/rl-global-mgmt/) and my own experience, I believe these should be falling into those individual buckets. 

Will do some of my own investigation / stats collection and see if I'm able to determine this concretely.

**Comment by monde** [comment]
Hi @ymylei . I double checked with Okta's core developer who is responsible for rate limiting at Okta's API level. Requests matching this example `/api/v1/apps/TESTID1/users` and `/api/v1/apps/TESTID2/users` are going to be accounted for in the same rate limit bucket for `/api/v1/apps/${id}`. Let me know if you can definitively measure this is not the case for your org. 

One thing to note for you and others reading this in the future is that API rate limits can be modified in a very granular level on your account by your Okta account manager via support ticket. So the rate limiting documentation is documenting the default rate limits and that is how the Okta Terraform Provider implements it's rate limiting solution.

In other words, imagine your application is hitting the `/api/v1/apps/${id}/users` endpoint in an extraordinary fashion starving out the rate limit for all other `/api/v1/apps/${id}/*` requests. You can put in a support request to have `/api/v1/apps/${id}/users` placed in its own rate limit bucket separate from all other `/api/v1/apps/${id}/*` requests.

I'm going to close this ticket for now. Let me know if you want to continue this discussion. Thanks!

**Comment by ymylei** [comment]
> I'm going to close this ticket for now. Let me know if you want to continue this discussion. Thanks!

@monde Thanks for the response and confirmation on the rate limiting behavior! This lines up with the documentation.

This issue isn't resolved since my primary point is the RL API Mutex does not account for the behavior this way. The referenced test is checking that it's accounted under the `app` bucket (`api/v1/apps` root) instead of the `app-id` bucket (`api/v1/apps/{id}`). This lines up with behavior I've seen, and causing things like the root API to be violated since the mutex is setting values returned for the wrong rate limit bucket.

With this information in mind, I can work on a PR to correct this behavior.

**Comment by monde** [comment]
Will be in `v3.20.4` release. Fixed in #933
